static/app/utils/useLocalStorageState.ts

```diff
@@ -119,7 +119,7 @@ function initializeStorage<S>(
 export function useLocalStorageState<S>(
   key: string,
   initialState: S | ((value?: unknown, rawValue?: unknown) => S)
-): [S, (value: S) => void] {
+): [S, (value: S | ((prevState: S) => S)) => void] {
```

Comment:
This not properly resembles useState from react, we werent supporting the callback version which accesses the previous state before